s3-exporter/templates/deployment.yaml

```diff
@@ -44,7 +44,7 @@ spec:
           {{- end }}
           envFrom:
           - secretRef:
-              name: {{ template "s3-exporter.bucketAccessSecretName" . }}
+              name: {{ .Values.s3.existingSecret | default (include "s3-exporter.fullname" .) }}
```

Comment:
I think you have a double logic here , where the secretName exists AND the existingSecret exists too.